### PR TITLE
feat: add search tags suggestions query

### PR DESCRIPTION
**Pull request by capJavert**
AS-281

Add query to use in search panel. https://github.com/dailydotdev/apps/pull/3067

Similar query is in `schema/keywords` but it is only for moderators and I wanted to expose this one from`schema/search` to match specific format we return for all of our search providers.

### CODE REVIEW

File: src/schema/search.ts [hunk 1]
```diff
@@ -312,6 +337,28 @@ export const resolvers: IResolvers<unknown, Context> = traceResolvers({
         query: args.query,
       };
     },
+    searchTagSuggestions: async (
```
Comment (rebelchris) [inline src/schema/search.ts:340]: I noticed this the other day why do we use keyword and tag differently 👀

File: src/schema/search.ts [hunk 2]
```diff
@@ -312,6 +337,28 @@ export const resolvers: IResolvers<unknown, Context> = traceResolvers({
...
+      source,
+      { query, limit }: SearchSuggestionArgs,
+      ctx,
+    ): Promise<GQLSearchSuggestionsResults> => {
+      const searchQuery = ctx.con
+        .getRepository(Keyword)
+        .createQueryBuilder()
+        .select(`value as id, COALESCE(flags->>'title', value) as title`)
+        .where(`status = 'allow'`)
+        .andWhere(`value ILIKE :query`, {
```
Comment (rebelchris) [inline src/schema/search.ts:350]: Well need to monitor performance on this one

**Review by rebelchris — APPROVED** [review]

File: src/schema/search.ts (hunk 1 above)
Comment (idoshamun) [reply src/schema/search.ts:340]: it's a legacy reason. At first we had tags and then i wanted to change the way i manage and discover the tags without affecting everything so I had to come up with a new name, keyword. Everything else was already using the tags term so I kept it. 🤷 
Bad decisions

File: src/common/search.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+export type SearchSuggestionArgs = {
+  query: string;
+  version: number;
+  limit?: number;
+};
+
+export const defaultSearchLimit = 3;
+export const maxSearchLimit = 100;
+
+export const getSearchLimit = ({
```
Comment (idoshamun) [inline src/common/search.ts:10]: You can simply do `Math.max(Math.min(limit, maxSearchLimit), 1)`

File: src/schema/search.ts [hunk 4]
```diff
@@ -312,6 +337,28 @@ export const resolvers: IResolvers<unknown, Context> = traceResolvers({
...
     },
+    searchTagSuggestions: async (
+      source,
+      { query, limit }: SearchSuggestionArgs,
+      ctx,
+    ): Promise<GQLSearchSuggestionsResults> => {
+      const searchQuery = ctx.con
+        .getRepository(Keyword)
+        .createQueryBuilder()
+        .select(`value as id, COALESCE(flags->>'title', value) as title`)
```
Comment (idoshamun) [inline src/schema/search.ts:348]: What is this title thingy?

**Review by idoshamun — APPROVED** [review]

File: src/schema/search.ts (hunk 1 above)
Comment (rebelchris) [reply src/schema/search.ts:340]: Ah so moving forward can we just start using keyword everywhere 👍

File: src/schema/search.ts (hunk 4 above)
Comment (rebelchris) [reply src/schema/search.ts:348]: I think its the flag that nimrod can set for sep purpose.
I think we should just search based on value (i think its called)

Comment (idoshamun) [reply src/schema/search.ts:348]: thanks for the explanation

**Review by nensidosari — APPROVED** [review]

File: src/common/search.ts (hunk 3 above)
Comment (capJavert) [reply src/common/search.ts:10]: Yep, nice!

File: src/schema/search.ts (hunk 1 above)
Comment (capJavert) [reply src/schema/search.ts:340]: I'll refactor so it is called keyword here, on frontend though we still use Tag almost everywhere.

Comment (idoshamun) [reply src/schema/search.ts:340]: I don't mind keeping it tags everywhere

Comment (capJavert) [reply src/schema/search.ts:340]: @rebelchris gonna merge this as is, let's sync on it tomorrow when you are back.

**Pull request merged**